doc/tutorials/optimizer/optimizer.txt

```diff
@@ -0,0 +1,251 @@
+/*!
+
+@file optimizer.txt
+@author Marcus Edel (https://kurg.org)
+@brief Tutorial for how to implement a new Optimizer in mlpack.
```

Comment:
I guess this should be `Tutorial on` instead of `Tutorial for`?

Reply:
Either is valid, in my opinion.